### PR TITLE
refactored build script

### PR DESCRIPTION
**Pull request by carocad**
- splitted several complex function into smaller pieces
- use of clojure edn reader for finding js/require statements
  - this is a preparation for us to be able to use [Clojurescript node modules interop](https://clojurescript.org/news/2017-07-12-clojurescript-is-not-an-island-integrating-node-modules)

This changes also had the unintended effect of speeding up the build process. In my machine it dropped from 60 seconds to 10 seconds :)
The reload time also seems a bit faster

This seems to work pretty good but it is still too early to be sure that it will work in all cases as there might some edge cases that I have not encounter. 

More testing is appreciated but I think we can solve those problems once they arrive :)

### CODE REVIEW

**Comment by mehdisadeghi** [comment]
:shipit: This branch has been deployed to:
  https://expo.io/@carocad/hive?release-channel=build-script

  Download the [Expo](https://expo.io/) app and scan this QR code to get started!

  QR Code

**Comment by mehdisadeghi** [comment]
:shipit: This branch has been deployed to:
  https://expo.io/@carocad/hive?release-channel=build-script

  Download the [Expo](https://expo.io/) app and scan this QR code to get started!

  QR Code

**Comment by mehdisadeghi** [comment]
:shipit: This branch has been deployed to:
  https://expo.io/@carocad/hive?release-channel=build-script

  Download the [Expo](https://expo.io/) app and scan this QR code to get started!

  QR Code

File: env/dev/user.clj [hunk 1]
```diff
@@ -4,11 +4,34 @@
...
   (:import (java.net NetworkInterface InetAddress Inet4Address)
-           (java.io File)))
+           (java.io File PushbackReader)))
 ;; This namespace is loaded automatically by nREPL
 
+(defn- read-clj
+  "read a clojure file into a sequence of edn objects.
+
+  WARNING:
+   EDN is a subset of Clojure code representations so it is not guarantee
```
Comment (mehdisadeghi) [inline env/dev/user.clj]: guaranteed*

Comment (carocad) [reply env/dev/user.clj]: Oh actually that description is no longer valid I have to update it 😅

**Comment by mehdisadeghi** [comment]
:shipit: This branch has been deployed to:
  https://expo.io/@carocad/hive?release-channel=build-script

  Download the [Expo](https://expo.io/) app and scan this QR code to get started!

  QR Code

**Pull request merged**